### ISSUE TITLE
bug: ConnectionError when using ELASTICSEARCH_BACKEND

### ISSUE THREAD

**Issue by Dgame** [localstack/localstack#5052]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

It's related to #4064 - I'm experiencing the same pain again. :)

I'm using the docker-compose.yml from the [tutorial](https://github.com/localstack/localstack/blob/master/doc/external_services_integration/elasticsearch/HOWTO.md) and just adjusted the SERVICE environment like this:
```
    environment:
      - SERVICES=es,firehose,s3
      - ELASTICSEARCH_BACKEND=http://elasticsearch:9200
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - DOCKER_HOST=unix:///var/run/docker.sock
```

So I removed some and added `s3` to the SERVICE. I can create an ES Domain like this:
```sh
aws --endpoint-url=http://localhost:4566 es create-elasticsearch-domain --domain-name es_local --elasticsearch-version 7.10 --elasticsearch-cluster-config '{ "InstanceType": "m3.xlarge.elasticsearch", "InstanceCount": 4, "DedicatedMasterEnabled": true, "ZoneAwarenessEnabled": true, "DedicatedMasterType": "m3.xlarge.elasticsearch", "DedicatedMasterCount": 3}'
```

but I cannot accessing `curl http://localhost:4571/_cluster/health` results in
> curl: (52) Empty reply from server

But `curl http://localhost:9200/_cluster/health` works...

Regardless, I can create a Firehose Stream like this:
```sh
aws --endpoint-url=http://localhost:4566 firehose create-delivery-stream --delivery-stream-name firehose_es_local_stream --delivery-stream-type DirectPut --elasticsearch-destination-configuration "RoleARN=arn:aws:iam::000000000000:role/Firehose-Reader-Role,DomainARN=arn:aws:es:us-east-1:000000000000:domain/es_local,IndexName=foo,S3Configuration={RoleARN=arn:aws:iam::000000000000:role/Firehose-Reader-Role,BucketARN=arn:aws:s3:::000000000000:kinesis-activity-backup-local,CompressionFormat=UNCOMPRESSED,EncryptionConfiguration={NoEncryptionConfig=NoEncryption},CloudWatchLoggingOptions={Enabled=false}},ProcessingConfiguration={Enabled=false}"
```

but as soon as I try to send something in it like this:

```sh
aws --endpoint-url=http://localhost:4566 firehose put-record --delivery-stream-name firehose_es_local_stream --record '{"Data":"eyAidGFyZ2V0IjogImJlcnJ5IiB9"}'
```

(See [this comment](https://github.com/localstack/localstack/issues/4064#issuecomment-852008779) for further info)

I get the error

> An error occurred (500) when calling the PutRecord operation (reached max retries: 2): <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
> <title>500 Internal Server Error</title>
> <h1>Internal Server Error</h1>
> <p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>

and in Docker I see this (short version only):

> elasticsearch.exceptions.ConnectionError: ConnectionError(MyHTTPConnectionPool(host='localhost', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f646432cb80>: Failed to establish a new connection: [Errno 111] Connection refused'))) caused by: ConnectionError(MyHTTPConnectionPool(host='localhost', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f646432cb80>: Failed to establish a new connection: [Errno 111] Connection refused')))

This is what confuses me most:
> (host='localhost', port=9200)

Why `localhost` and not my external Cluster?
It seems like the registered `- ELASTICSEARCH_BACKEND=http://elasticsearch:9200` is not identified. I even tried my "old trick" from the old issue and used

```yml
      - ELASTICSEARCH_BACKEND=http://elasticsearch
      - ELASTICSEARCH_PORT=9200
```

but the result was the same...

### Expected Behavior

Well, Im just want to send data to my external Elasticsearch Cluster... 😅 

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

`docker compose up -d --build --remove-orphans` with the `docker-compose.yml` from the [tutorial](https://github.com/localstack/localstack/blob/master/doc/external_services_integration/elasticsearch/HOWTO.md).

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

Create an ES Domain like this:
```sh
aws --endpoint-url=http://localhost:4566 es create-elasticsearch-domain --domain-name es_local --elasticsearch-version 7.10 --elasticsearch-cluster-config '{ "InstanceType": "m3.xlarge.elasticsearch", "InstanceCount": 4, "DedicatedMasterEnabled": true, "ZoneAwarenessEnabled": true, "DedicatedMasterType": "m3.xlarge.elasticsearch", "DedicatedMasterCount": 3}'
```

Create a Firehose Stream like this:
```sh
aws --endpoint-url=http://localhost:4566 firehose create-delivery-stream --delivery-stream-name firehose_es_local_stream --delivery-stream-type DirectPut --elasticsearch-destination-configuration "RoleARN=arn:aws:iam::000000000000:role/Firehose-Reader-Role,DomainARN=arn:aws:es:us-east-1:000000000000:domain/es_local,IndexName=foo,S3Configuration={RoleARN=arn:aws:iam::000000000000:role/Firehose-Reader-Role,BucketARN=arn:aws:s3:::000000000000:kinesis-activity-backup-local,CompressionFormat=UNCOMPRESSED,EncryptionConfiguration={NoEncryptionConfig=NoEncryption},CloudWatchLoggingOptions={Enabled=false}},ProcessingConfiguration={Enabled=false}"
```

And put some data in the stream like this:

```sh
aws --endpoint-url=http://localhost:4566 firehose put-record --delivery-stream-name firehose_es_local_stream --record '{"Data":"eyAidGFyZ2V0IjogImJlcnJ5IiB9"}'
```

### Environment

```markdown
- OS: WSL (Ubuntu 20.04) on Windows 10
- LocalStack: latest
```

### Anything else?

It's related to #4064 - I'm experiencing the same pain again. :)

**Comment by Dgame** [comment]
I tried down to 0.12.10 and there I get this errors as soon as I try to `curl http://localhost:4571/_cluster/health`:

```
ERROR: 'su localstack -c '/tmp/localstack/elasticsearch/7.7.0/bin/elasticsearch -E http.port=53973 -E http.publish_port=53973 -E http.compression=false -E path.data=/tmp/localstack/elasticsearch/7.7.0/data -E path.repo=/tmp/localstack/es_backup -E xpack.ml.enabled=false'': exit code 1; output: b"OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.\n[2021-12-01T22:00:29,682][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [9788c24e0ac7] uncaught exception in thread [main]\norg.elasticsearch.bootstrap.StartupException: java.lang.IllegalStateException: Unable to access 'path.repo' (/tmp/localstack/es_backup)\n\tat org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:174)
[...]
MyHTTPConnectionPool(host='elasticsearch', port=80): Max retries exceeded with url: /_cluster/health (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe037056670>: Failed to establish a new connection: [Errno -3] Try again'))
```

but well, at least the `host` is `elasticsearch` and not `localhost` anymore. :D But port 80 seems still strange.

After 0.12.10 until 0.13.0 I get `Created: false` if I try to create a elasticsearch domain which is different from the usual `Created: true`. 

If I try `curl http://localhost:4571/_cluster/health` in 0.12.10 (and with `- ELASTICSEARCH_BACKEND=http://elasticsearch:9200`), I get the error with the correct port **and** host (but still an error):

`MyHTTPConnectionPool(host='elasticsearch', port=9200): Max retries exceeded with url: /_cluster/health`

But if I try to put data into the stream the error is

`ConnectionError(MyHTTPConnectionPool(host='localhost', port=9200)): Max retries exceeded with url: /foo/_create/91773b1e-49d9-4220-a723-`

so the host is wrong again.

**Comment by thrau** [comment]
hi Randy @Dgame, sorry about the confusion there, the `ELASTICSEARCH_BACKEND` is pretty much defunct. we've re-written the way that elasticsearch backends are managed, and currently there's now way of configuring your own. here's the doc on how it currently works: https://docs.localstack.cloud/aws/elasticsearch/ which should also clear up some of the confusion around the port that elasticsearch is exposed on.

do you need to add your own elasticsearch backend? or can you use the one that localstack starts (which is any elasticsearch version >= 5)?

**Comment by Dgame** [comment]
Well, ideally I would like to use our own backend since we use this particular cluster for other tasks as well. Is there any way I can help with this, or is there an estimate of when it will be working again? Or maybe I (we as a company) need to use the Pro version?

**Comment by thrau** [comment]
i see. no problem, i'll take a look, shouldn't be that much effort to re-implement.

**Comment by thrau** [comment]
Hey Randy @Dgame , we've released a fix with #5073 that's deployed in the `latest` tag of the localstack container image. It re-introduces the custom backend behavior that you were looking for. There are a few changes: the variable is now called `ES_CUSTOM_BACKEND`, and the cluster endpoints are created with the `domain` `ES_ENDPOINT_STRATEGY`. please find the updated docs at https://docs.localstack.cloud/aws/elasticsearch/#custom-elasticsearch-backends.

please let us know if this works for you, so i can close the issue.

**Comment by Dgame** [comment]
@thrau It seems to work now! Thanks!
